crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -180,42 +86,53 @@ pub struct CheckpointExecutorEventLoop {
     metrics: Arc<CheckpointExecutorMetrics>,
 }
 
-impl CheckpointExecutorEventLoop {
-    fn new(
+impl CheckpointExecutor {
+    pub fn new(
         mailbox: broadcast::Receiver<VerifiedCheckpoint>,
-        end_of_epoch_event_sender: broadcast::Sender<Committee>,
         checkpoint_store: Arc<CheckpointStore>,
         authority_state: Arc<AuthorityState>,
         config: CheckpointExecutorConfig,
-        metrics: Arc<CheckpointExecutorMetrics>,
+        prometheus_registry: &Registry,
     ) -> Result<Self, TypedStoreError> {
         Ok(Self {
             mailbox,
-            end_of_epoch_event_sender,
             checkpoint_store,
             authority_state,
             config,
             highest_scheduled_seq_num: None,
             latest_synced_checkpoint: None,
             end_of_epoch: false,
-            metrics,
+            metrics: CheckpointExecutorMetrics::new(prometheus_registry),
```

Comment:
Would recreating this each epoch lead to issues? I remember that registry prevents you from registering the same metric twice

Reply:
oh good callout. I think you're right

Reply:
Yeah we'll want these metrics to be created a single time and then probably need to have them passed in on each incarnation

Reply:
This seems not updated

Reply:
It is indeed not updated. Because we are no longer re-instantiating executor, therefore the problem no longer exists.

Reply:
Oh I see. So this PR no longer makes executor a per-epoch component, just that we get rid of the channel and replaced it with call-return.